extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/internal/NbProjectInfoAction.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.gradle.tooling.internal;
+
+import java.io.Serializable;
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.BuildController;
+
+/**
+ *
+ * @author lkishalmi
+ */
+class NbProjectInfoAction implements Serializable, BuildAction<NbProjectInfo>{
```

Comment:
not sure I understand the title. I see that NbProjectInfoAction is in the module which is set to release=17. Where is the magic that it makes this class release=8?

Reply:
oh, there it is:
```xml
    <target name="compile" depends="prepare-libs">
        <mkdir dir="build/classes/java/main"/>
        <javac srcdir="src/main/java" destdir="build/classes/java/main" classpathref="compile.classpath" release="8" includeantruntime="false" debug="true"/>
    </target>
```

Reply:
well, I was confused until I looked through the module and found the snippet I pasted. This class is _not_ compiled as part of the module and uses a different lang level. So everything is fine.

I would try to extract the `8` into a `tooling.javac.release=8` property so that it is next to the module property. Comment to keep it as low as possible would be great too.

Reply:
ah this uses different properties. In any case, a comment somewhere would be good ;)

Reply:
Moved some properties into the `project.properties`, added some comments there as well.